kola/harness.go

```diff
@@ -460,19 +460,21 @@ func SetupOutputDir(outputDir, platform string) (string, error) {
 	}
 
 	if defaulted {
+		tempLinkPath := filepath.Join(outputDir, "latest")
 		linkPath := filepath.Join(defaultBaseDirName, platform+"-latest")
+		// safety check
 		st, err := os.Lstat(linkPath)
-		if err == nil {
-			if (st.Mode() & os.ModeType) != os.ModeSymlink {
-				return "", fmt.Errorf("%v exists and is not a symlink", linkPath)
-			}
-			if err := os.Remove(linkPath); err != nil {
-				return "", err
-			}
-		} else if !os.IsNotExist(err) {
+		if err == nil && (st.Mode()&os.ModeType) != os.ModeSymlink {
+			return "", fmt.Errorf("%v exists and is not a symlink", linkPath)
+		} else if err != nil && !os.IsNotExist(err) {
+			return "", err
+		}
+		if err := os.Symlink(defaultDirName, tempLinkPath); err != nil {
```

Comment:
Currently `tempLinkPath` is only removed if the Rename succeeds, is this intended or should we move it to a defer'd cleanup?

Reply:
It's being removed only if the rename fails.  If it succeeds, the link is now at its final location so we don't need to clean it up.

Reply:
Whoops, brain not working this morning need more caffeine